sdk/src/Core/Amazon.Runtime/Internal/Util/Logger.cs

```diff
@@ -80,39 +77,76 @@ private void ConfigureLoggers()
                     il.IsEnabled = (logging & LoggingOptions.Console) == LoggingOptions.Console;
                 if (il is InternalSystemDiagnosticsLogger)
                     il.IsEnabled = (logging & LoggingOptions.SystemDiagnostics) == LoggingOptions.SystemDiagnostics;
+
+                if (!IsEnabled)
```

Comment:
For all of these properties the last `InternalLogger` will overwrite the value on the `Logger`. If I understand the purpose the properties on the `Logger` should be true if any of the `InternalLogger` are true.

Reply:
I do suck at boolean logic and would probably fail any job interview 👯‍♂️ 

Isn't that exactly what it is doing? If the property is false it will check the property of the internal logger and set it to the corresponding value of the internal logger. If the internal logger is true it sets the property to true and subsequent checks don't need to look at the internal logger value anymore. If the property gets set again to false the next iteration needs to check the value of the next internal logger. So if all are false it remains false. If one is true it will turn to true and subsequent checks omit toggling it. I wanted to avoid 'Any()'